gears/src/x/ante.rs

```diff
@@ -142,6 +146,56 @@ impl<AK: AuthKeeper<SK>, BK: BankKeeper<SK>, SK: StoreKey, GC: SignGasConsumer>
         Ok(())
     }
 
+    fn mempool_fee<M: TxMessage, DB: Database>(
+        &self,
+        ctx: &mut TxContext<'_, DB, SK>,
+        TxWithRaw {
+            tx,
+            raw: _,
+            tx_len: _,
+        }: &TxWithRaw<M>,
+    ) -> anyhow::Result<()> {
+        if !ctx.is_check() {
+            return Ok(());
+        }
+
+        let fee = tx.auth_info.fee.amount.as_ref();
+        let gas = tx.auth_info.fee.gas_limit;
+
+        let min_gas_prices = ctx.options.min_gas_prices();
+
+        if min_gas_prices.is_empty() || min_gas_prices.iter().any(|this| this.amount.is_zero()) {
+            return Ok(());
+        }
+
+        if let Some(fee_coins) = fee {
+            let mut required_fees = Vec::with_capacity(min_gas_prices.len());
+
+            for gp in min_gas_prices {
+                required_fees.push(Coin {
+                    denom: gp.denom,
+                    amount: gp
+                        .amount
+                        .checked_mul_ceil(cosmwasm_std::Decimal::new(u64::from(gas).into()))?, // TODO:NOW Great. Replace it with something sane
+                });
+            }
+
+            if fee_coins.inner().into_iter().any(|fee_coin| {
+                required_fees
+                    .iter()
+                    .any(|req_fee| req_fee.amount >= fee_coin.amount)
```

Comment:
Couldn't these be different denominations? For every required fee we need to check that the fee contains at least that much of that denomination.

Reply:
You're right, I found [where](https://github.com/cosmos/cosmos-sdk/blob/d3f09c222243bb3da3464969f0366330dcb977a8/types/coin.go#L572) it does it

Reply:
OK, so we must do the same